### DISCUSSION TITLE
Wiring of dro

### DISCUSSION THREAD

**Discussion by Team-Deepflights** [phil-barrett/grblHAL-teensy-4.x#103] [General]
Hi Phil,
Finally I finished the mill and I would like to add a MPG for easier handling.
The serial 2 input is used for WIFI connection, so it can’t de used as input.
I found one at Ebay that would fit my needs.

I would like to use this one: 

https://www.ebay.de/itm/403501037361?mkcid=16&mkevt=1&mkrid=707-127634-2357-0&ssspo=mmlr5MygSqq&sssrc=4429486&ssuid=kqkWdZmoTVG&var=&widget_ver=artemis&media=COPY

Do I need an Arduino between and go over I2C for use it or could it be wired directly?
If yes, do you please provide a wiring diagram?

I tried to find some information about that but the results are not very useful.
Also I found a wireless one which use a dongle but I think this is a question that should be asked at the GBRLhal repo.

BTW, here is a photo of the mill, A-axis is still on the way.

![image](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/96638334/85b5ef5a-df7d-4882-904f-e8933fb0d84d)

**Comment by phil-barrett** [comment]
That is a very nice looking machine!

Not wanting to split hairs but the thing you linked to is not a DRO (digital readout) but rather a "Pendant", "Dongle" or "MPG". It can not be directly hooked up to the breakout board.  However, I am working on a project to support a Pendant/dongle/MPG.  It entails a small microcontroller to connect to a grblHAL breakout board. It will support either I2C or UART.  The current designs use an RP2040 or Rasp Pi Pico.  Here is a rendering of the first board I have made. It is designed to plug directly onto the I2C header of my breakout boards. It will support either DB15 or wire only versions of the pendant.
![PR2040-MPG](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/16570958/467141c4-db4b-46be-97b4-4a1c13fe2afd)
My first build is non-functional (it doesn't talk USB) so I'm debugging. I also have a Pico based version that I am using to get the SW running.  Here's a render of that one.  I have it talking to the pendant via the DB15 connector. Next I need to get it talking to grblHAL.
![test2](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/16570958/a5b60ca8-0b3f-4829-9dbf-43ea56c8c93c)

A long vacation trip followed by a nasty bout of flu (or whatever I picked up in Argentina) slowed down progress but I'm getting back to it.  This [repo](https://github.com/phil-barrett/PicoPendant) is where I will put things.

**Comment by phil-barrett** [comment]
If you want to get going sooner, you might want to look at using an xbox controller with the GCode Sender of your choice. Use key mapping software like Keysticks to map the controller buttons into keystrokes to send to the GCode Sender. I have one working with ioSender. Works pretty well.

**Reply by Team-Deepflights** [reply]
This seem that the wireless pendant (thx for explaining) 😉 is the way to go.
I bookmarked the repo, sounds very nice.

Thank you, the mill is now #4, a summary of experience.
I’m going to make a U-Tube vid about it, maibe some ideas inspire others.

**Reply by phil-barrett** [reply]
Great, looking forward to it.  Will be happy to link to your vid when you get it done so others can gain from your efforts. Wired controllers also work, btw.

**Reply by Team-Deepflights** [reply]
Do you have some example for a wired controller? my laptop dont have any RS232 plug. I am limited to USB or wireless.
The Xbox controller sound nice, but I prefer something that is closer to a real controller, if you know what I mean.

**Reply by phil-barrett** [reply]
Sorry, don't have one that isn't a game controller.  There are USB MPGs but I haven't played with them at all.
